src/util/triplea/MapXMLCreator/GameSequencePanel.java

```diff
@@ -0,0 +1,186 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map.Entry;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class GameSequencePanel extends DynamicRowsPanel {
+	
+	public GameSequencePanel(final JPanel stepActionPanel)
+	{
+		super(stepActionPanel);
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		if (s_me == null || !(s_me instanceof GameSequencePanel))
+			s_me = new GameSequencePanel(stepActionPanel);
+		DynamicRowsPanel.layout(mapXMLCreator, stepActionPanel);
+	}
+	
+	protected ActionListener getAutoFillAction()
+	{
+		return new ActionListener() {
+			
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				if (JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(m_ownPanel, "Are you sure you want to use the  Auto-Fill feature?\r\nIt will remove any information you have entered in this step and propose commonly used choices.", "Auto-Fill Overwrite Warning", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE)) {
+					MapXMLHelper.s_gamePlaySequence.clear();

```

Comment:
Lots of static state. MapXMLHelper looks like a good candidate for the builder pattern.